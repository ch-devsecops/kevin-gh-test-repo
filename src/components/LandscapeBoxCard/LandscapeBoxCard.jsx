import React from 'react';
import PropTypes from 'prop-types';

import {
  Wrapper,
  Column,
  Box,
  Markdown,
  MarkdownHeading,
  MediaItem,
  useOptionalVideo,
  Media,
} from '@honda-canada/design-system-react';

import themeStyles from './LandscapeBoxCard.styles';

const Container = themeStyles.apply(Box, 'Container');
const MediaContainer = themeStyles.apply(Box, 'MediaContainer');
const CTAGroup = themeStyles.apply(Box, 'CTAGroup');
const ContentBox = themeStyles.apply(Box, 'ContentBox');

const LandscapeBoxCard = ({
  title,
  bodyText,
  backgroundColor,
  contentCardColor,
  contentAlignment,
  desktopImage,
  video,
  ctas,
  isTertiaryCTAGroup,
  anchorId,
  mobileImage,
  imageTitle,
  gtmTags = {},
  isDark,
}) => {
  const { optionalVideo } = useOptionalVideo(video);

  return (
    <Box
      bg={isDark ? 'black' : backgroundColor}
      py={['l', 'big']}
      id={anchorId}
      data-gtm-component-type={gtmTags.type}
      data-gtm-category={gtmTags.category}
    >
      <Wrapper gutters={[false]}>
        <Column width={1}>
          <Container contentAlignment={contentAlignment}>
            <ContentBox bg={isDark ? '#000000' : contentCardColor}>
              {imageTitle ? (
                <Box px={[0, 'default']} mb={['xs', 'm']}>
                  {imageTitle}
                </Box>
              ) : (
                <MarkdownHeading
                  px={[0, 'default']}
                  mb={['xs', 'm']}
                  style={{ textAlign: 'center' }}
                  color={isDark ? 'white' : undefined}
                  headingOverride="h3"
                >
                  {title}
                </MarkdownHeading>
              )}
              <Box px={['m', 'xxl']}>
                <Markdown style={{ textAlign: 'center' }} mb={[0, 0]} color={isDark ? 'white' : undefined}>
                  {bodyText}
                </Markdown>
              </Box>
              <CTAGroup isTertiary={isTertiaryCTAGroup}>{ctas}</CTAGroup>
            </ContentBox>

            <Media greaterThanOrEqual="smallDesktop">
              <MediaContainer contentAlignment={contentAlignment}>
                <MediaItem image={desktopImage} video={optionalVideo} containerSize="100%" />
              </MediaContainer>
            </Media>
            <Media lessThan="smallDesktop">
              <MediaContainer contentAlignment={contentAlignment}>
                <MediaItem image={mobileImage || desktopImage} video={optionalVideo} containerSize="100%" />
              </MediaContainer>
            </Media>
          </Container>
        </Column>
      </Wrapper>
    </Box>
  );
};

LandscapeBoxCard.defaultProps = {
  contentCardColor: 'grey.5',
  backgroundColor: 'white',
  bodyText: '',
  contentAlignment: 'left',
  isTertiaryCTAGroup: false,
};

LandscapeBoxCard.propTypes = {
  title: PropTypes.string,
  /**
   * Markdown enabled text
   */
  bodyText: PropTypes.string,
  backgroundColor: PropTypes.oneOf(['white', 'grey.5']),
  contentCardColor: PropTypes.oneOf(['white', 'grey.5']),
  /**
   * If true, `contentCardColor` is replaced with `#000000` and text is white
   */
  isDark: PropTypes.bool,
  contentAlignment: PropTypes.oneOf(['left', 'right']),
  desktopImage: PropTypes.element,
  video: PropTypes.shape({
    onPlay: PropTypes.func,
  }),
  ctas: PropTypes.arrayOf(PropTypes.element),
  isTertiaryCTAGroup: PropTypes.bool,
  anchorId: PropTypes.string,
  mobileImage: PropTypes.element,
  gtmTags: PropTypes.shape({
    category: PropTypes.string,
    type: PropTypes.string,
  }),
  imageTitle: PropTypes.element,
};

export default LandscapeBoxCard;
