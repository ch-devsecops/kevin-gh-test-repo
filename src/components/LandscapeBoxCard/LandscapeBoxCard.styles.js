import css from '@styled-system/css';
import variant from '@styled-system/variant';
import color from '@styled-system/color';

import { ThemeStyles } from '@honda-canada/design-system-react';

const themeStyles = new ThemeStyles();

themeStyles.addBaseStyles('MediaContainer')(({ contentAlignment }) => {
  const baseAttrs = {
    position: 'absolute',
    maxWidth: ['848px'],
    width: ['100%', '67%'],
    height: ['260px', '477px'],
    top: 0,
  };

  if (contentAlignment === 'right') {
    baseAttrs.left = 0;
  } else {
    baseAttrs.right = 0;
  }

  return css(baseAttrs);
});

themeStyles.addBaseStyles('Container')(({ contentAlignment }) => {
  const desktopFlexDirection = contentAlignment === 'right' ? 'row-reverse' : 'row';

  return css({
    display: 'flex',
    position: 'relative',
    minHeight: ['260px', '477px'],
    flexDirection: ['column', desktopFlexDirection],
    justifyContent: ['center', 'flex-start'],
    '&:after': {
      content: '""',
      minHeight: [0, 'inherit'],
    },
  });
});

themeStyles.addBaseStyles('ContentBox')(
  css({
    backgroundColor: 'white',
    height: ['315px', '408px'],
    display: 'flex',
    flexDirection: 'column',
    alignContent: 'center',
    justifyContent: 'center',
    alignSelf: 'center',
    zIndex: '1',
    '& > *': {
      alignSelf: 'center',
    },
    marginTop: ['228px', 0],
    width: ['calc(100% - 20px)', '350px', '466px'],
  }),
  color,
  variant({
    prop: 'styling',
    variants: {
      nsx: {
        position: 'absolute',
        right: '0',
        bottom: '0',
        paddingLeft: ['0', '85px'],
        marginRight: ['0', '12px'],
        width: ['100%', '530px'],
        height: ['315px', '100%'],
        background: ['transparent', 'linear-gradient(270deg, #000000 0%, rgba(0, 0, 0, 0) 100%)'],
      },
    },
  }),
);

themeStyles.addBaseStyles('CTAGroup')(({ isTertiary }) =>
  css({
    display: 'flex',
    flexDirection: 'column',
    alignItems: 'center',
    mt: isTertiary ? ['m', 'l'] : 'default',
    '& > *': {
      mb: isTertiary ? ['m', 'l'] : 'm',
    },
    '& >:last-child': {
      mb: 0,
    },
  }),
);

export default themeStyles;
