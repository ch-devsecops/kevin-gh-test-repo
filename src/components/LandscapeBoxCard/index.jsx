import React, { useContext } from 'react';
import { Image } from '@honda-canada/design-system-react';
import LandscapeBoxCardReact from './LandscapeBoxCard';
import { getVideoProps, mapGTMCategory, mapJssFieldsToCtaComponents } from '../../utils/sitecoreFields';
import { ModelExplorationContext } from '../ModelExplorationContext';
import { useAppName } from '../../utils/sitecoreContext';
import { PSP_SITE_NAME } from '../../utils/constants';

const LandscapeBoxCard = ({ fields, params, rendering }) => {
  const modelExplorationContext = useContext(ModelExplorationContext) || {};
  const appName = useAppName();
  const { isDark } = modelExplorationContext;

  if (!fields) {
    return null;
  }
  const { anchorId, bodyText, ctaType1, desktopImage, gtmCategory, gtmTitle, mobileImage, title, videoUrl, nameBadge } =
    fields;

  let componentType;

  switch (appName) {
    case PSP_SITE_NAME:
      fields.gtmInteractionType1 = { value: 'cta: explore' };
      fields.gtmInteractionType2 = { value: 'cta: explore' };
      fields.gtmInteractionType3 = { value: 'cta: explore' };
      componentType = rendering?.componentName;
      break;
    default:
      break;
  }

  // componentType was not applied to the CTA in auto but is required for PSP as stated in ticket issu SCC-5158
  const ctas = mapJssFieldsToCtaComponents(fields, gtmTitle?.value, componentType);
  const gtmTags = {
    category: mapGTMCategory(gtmCategory),
    title: gtmTitle?.value,
    type: rendering?.componentName,
  };

  return (
    <LandscapeBoxCardReact
      imageTitle={nameBadge?.value?.src ? <Image {...nameBadge.value} /> : null}
      title={title?.value}
      bodyText={bodyText?.value}
      backgroundColor={params?.bgColour === 'Grey' ? 'grey.5' : 'white'}
      contentCardColor={params?.contentBgColour === 'Grey' ? 'grey.5' : 'white'}
      contentAlignment={params?.contentAlignment?.toLowerCase()}
      desktopImage={desktopImage?.value?.src && <Image {...desktopImage.value} />}
      mobileImage={mobileImage?.value?.src && <Image {...mobileImage.value} />}
      video={getVideoProps(videoUrl?.value)}
      ctas={ctas}
      isTertiaryCTAGroup={ctaType1?.value === 'Tertiary'}
      anchorId={anchorId?.value}
      gtmTags={gtmTags}
      isDark={isDark}
    />
  );
};

export default LandscapeBoxCard;
