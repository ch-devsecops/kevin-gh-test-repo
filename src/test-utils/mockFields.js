/* eslint-disable max-len */
/* eslint-disable max-classes-per-file */

// FIXME: jsdoc popup not working on test files
/** Class representing a mock data. */
class MockData {
  /**
   * Add a field.
   * @param {string} key - key of the field obj: i.e. 'title'
   * @param {Object} value - value of the field obj: i.e. { value: 'some text' }.
   */
  add(key, value) {
    this[key] = value;
  }
}

/**
 * Class representing mock fields object.
 * @extends MockData
 *
 * available method: add()
 *
 * default fields: [anchorId, title, bodyText, image, ctaLink, ctaType, ctaIcon, gtmCategory, gtmTitle, contentTitle, subtitle]
 *
 * note: we can add more commonly used fields in the class declaration if needed
 */
class MockFields extends MockData {
  constructor() {
    super();

    this.title = { value: 'Title' };
    this.bodyText = { value: 'BodyText' };
    this.image = {
      value: { src: 'mock.png', alt: 'mock alt' },
    };
    this.ctaLink = {
      value: {
        href: 'www.source.ca',
        title: 'cta aria label',
        text: 'CTA Text',
      },
    };
    this.ctaType = {
      value: 'Tertiary',
    };
    this.ctaIcon = {
      fields: {
        value: {
          value: 'arrowRight',
        },
      },
    };
    this.gtmCategory = {
      fields: {
        value: {
          value: 'Others',
        },
      },
    };
    this.gtmTitle = {
      value: 'gtm title',
    };
    this.contentTitle = {
      value: 'Content Title',
    };
    this.subtitle = {
      value: 'Content Subtitle',
    };
    this.seoH1 = {
      value: '',
    };
    this.anchorId = {
      value: 'anchor-id',
    };
    this.ctaLabel = {
      value: 'CTA Label',
    };
  }
}

/**
 * Class representing a mock rendering object
 * @extends MockData
 *
 * available method: add()
 */
class MockRendering extends MockData {
  /**
   * Create a mock rendering object
   * @param {string} componentName - adds componentName
   */
  constructor(componentName) {
    super();

    this.componentName = componentName;
  }
}

export { MockFields, MockRendering };
