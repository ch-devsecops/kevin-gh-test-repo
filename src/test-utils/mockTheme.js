/**
 * Mock styled-components theme, if you want to test how your component
 * implements theme styles. Import as-is, or spread in any
 * additional values you need.
 *
 * Example usage:

  it('should apply the bgColor param', () => {
    const { colors } = MockTheme;
    const { container } = render(
      <MyComponent
        fields={mockFields}
        params={{ bgColor: 'Gray' }}
      />,
    );
    const wrapper = container.querySelector('section');

    expect(wrapper).toHaveStyleRule('background-color', colors.grey[5]);
  });
 */

import { HONDA_THEME_NAME } from '../utils/constants';

const mockTheme = {
  letterSpacings: {
    sm: '0.5px',
    md: '1px',
  },
  lineHeights: {
    sm: '22px',
    md: '26px',
  },
  colors: {
    white: '#fff',
    transparent: 'transparent',
    grey: ['#707070', '#C7C7C7', '#D5D5D5', '#EAEAEA', '#F8F8F8', '#F5F5F5'],
    black: '#2C2C2C',
    red: '#DA2C2E',
    redShades: ['rgba(200, 16, 46, 0.1)'],
    aodaRed: '#D92629',
    lightRed: '#EC9597',
    darkRed: '#A21C1D',
    blue: '#334C96',
    darkBlue: '#002D62',
    lightBlue: '#00447826',
    yellow: ['#F9CA52', '#EDAB00', '#8D763F'],
    fadedBlue: '#ADB7D5',
    primary: '#DA2C2E',
    primaryAoda: '#D92629',
    primaryDark: '#A21C1D',
    primaryLight: '#EC9597',
    secondary: '#2C2C2C',
    secondaryLight: '#C7C7C7',
    tertiary: '#002D62',
    tertiaryLight: '#334C96',
    tertiaryDisabled: '#ADB7D5',
    danger: '#DA2C2E',
    warning: '#2C2C2C',
    warningLight: '#EAEAEA',
    success: '#334C96',
    successLight: '#d9e0e7',
    disabled: '#C7C7C7',
    typographyDefault: '#2C2C2C',
  },
  space: [0, 4, 8, 12, 16, 24, 32, 48, 64, 96, 128, 144, 192, 256],
  borders: [0, '1px solid', '2px solid', '3px solid'],
  breakpoints: ['768px', '992px', '1200px'],
  fontWeights: {
    regular: 'normal',
    bold: 700,
  },
  fontSizes: {
    sm: '12px',
  },
  shadows: {
    smooth: {
      md: '0.1px 0.3px 0.3px rgba(0, 0, 0, 0.02),0.3px 0.7px 0.9px rgba(0, 0, 0, 0.028),0.7px 1.4px 1.8px rgba(0, 0, 0, 0.035),1.5px 2.9px 3.7px rgba(0, 0, 0, 0.047),4px 8px 10px rgba(0, 0, 0, 0.07)',
    },
    box: {
      subNav: '0px 2px 4px rgba(0, 0, 0, 0.1)',
    },
  },
  zIndices: {
    accordionItemTitle: 200,
    modal: 13001,
    sectionNav: 13000,
  },
  transitionDuration: {
    t1: '1s',
  },
  transitionTimingFunction: {
    default: 'ease-in-out',
    out: 'ease-out',
    in: 'ease-in',
  },
  name: HONDA_THEME_NAME,
  fonts: {
    default: 'Helvetica',
    bold: 'Cooper Black',
    heading: 'Roboto',
  },
  header: {
    desktop: {},
    mobile: {},
  },
  icons: {},
  logIconWarnings: false,
};

export default mockTheme;
