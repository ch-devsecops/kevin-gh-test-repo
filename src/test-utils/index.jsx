import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import { SitecoreContext } from '@sitecore-jss/sitecore-jss-react';
import { BrowserRouter } from 'react-router-dom';
import { OptionalVideoModalContextProvider } from '@honda-canada/design-system-react';
import UserLocationProvider from '@honda-canada/user-location';
import { configureAxe } from 'jest-axe';
import ModelExplorationContextProvider from '../components/ModelExplorationContext';
import ModelFiltersContextProvider from '../components/ModelFiltersContext';
import { MockFields, MockRendering } from './mockFields';
import MockTheme from './mockTheme';
import 'intersection-observer';
import { ACURA_THEME_NAME } from '../utils/constants';

const generateJssWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <SitecoreContext
        context={{
          language: 'en',
          site: { name: ACURA_THEME_NAME },
          ...context?.sitecoreContext,
        }}
        componentFactory={{}}
      >
        <BrowserRouter>
          <UserLocationProvider>
            <ThemeProvider theme={context?.theme || MockTheme}>
              <OptionalVideoModalContextProvider>
                <ModelExplorationContextProvider>
                  <ModelFiltersContextProvider>{children}</ModelFiltersContextProvider>
                </ModelExplorationContextProvider>
              </OptionalVideoModalContextProvider>
            </ThemeProvider>
          </UserLocationProvider>
        </BrowserRouter>
      </SitecoreContext>
    );
  };
/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in an typical JSS app's context providers.
 * Note that you can include a `context` object to set or override values
 * for SitecoreContext and ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.context] - default values for SitecoreContext, ThemeProvider
 * @param {string} [options.context.sitecoreContext] - value for SitecoreContext
 * @param {string} [options.context.theme] - value for ThemeProvider
 * @example
 * const { getByText } = jssRender(
 *  <MyComponent />, { context: { sitecoreContext: { language: 'fr' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const jssRender = (component, options = {}) => {
  const { context } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.context;

  return render(component, {
    wrapper: generateJssWrapper(context),
    ...options,
  });
};

const generateWrapper = context =>
  // eslint-disable-next-line react/display-name
  function ({ children }) {
    return (
      <BrowserRouter>
        <ThemeProvider theme={context?.theme || MockTheme}>
          <OptionalVideoModalContextProvider>{children}</OptionalVideoModalContextProvider>
        </ThemeProvider>
      </BrowserRouter>
    );
  };

/**
 * A custom implementation of the testing-library `render` function that
 * wraps tests in typical design-system context providers.
 * Note that you can include a `theme` object to set or override values
 * for ThemeContext.
 * @param {ReactElement} component - the component you want testing-library to render
 * @param {object} [options] - see https://testing-library.com/docs/react-testing-library/api/#render-options
 * @param {string} [options.theme] - default value for ThemeProvider
 * @example
 * const { getByText } = render(
 *  <MyComponent />, { theme: { space: { wrapperGap: '15px' } } }
 * );
 * @returns {function(): void} custom testing-library render function
 */
const customRender = (component, options = {}) => {
  const { theme } = options;
  // eslint-disable-next-line no-param-reassign
  delete options.theme;

  return render(component, {
    wrapper: generateWrapper(theme),
    ...options,
  });
};

// disable accessibility checks for errors that are usually trigged by content mistakes
const axe = configureAxe({
  rules: {
    'duplicate-id': { enabled: false },
    'heading-order': { enabled: false },
  },
});

export * from 'jest-axe';
export { axe };
export * from '@testing-library/react';
export { customRender as render, jssRender };
export { MockFields, MockRendering, MockTheme };
