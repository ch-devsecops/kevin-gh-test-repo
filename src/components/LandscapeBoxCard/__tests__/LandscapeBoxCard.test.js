import React from 'react';
import { axe, cleanup, render, MockRendering, MockFields } from '../../../test-utils';
import LandscapeBoxCard from '../LandscapeBoxCard';

afterEach(cleanup);

const mockFields = new MockFields();
const mockRendering = new MockRendering('LandscapeBoxCard');

describe('<LandscapeBoxCard>', () => {
  it('should render correctly with all props', async () => {
    const { container } = render(
      <LandscapeBoxCard fields={mockFields} rendering={mockRendering} title="My title" bodyText="My body text" />,
    );

    expect(await axe(container)).toHaveNoViolations();
    expect(container).toMatchSnapshot();
  });
});
