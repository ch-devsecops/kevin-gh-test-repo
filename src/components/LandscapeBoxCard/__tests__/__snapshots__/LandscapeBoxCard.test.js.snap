// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LandscapeBoxCard> should render correctly with all props 1`] = `
<div>
  <div
    class="Box-sc-1bsf784-0 lcphFz"
  >
    <div
      class="Box-sc-1bsf784-0 Wrapper-n67dtg-0 gXilRl"
    >
      <div
        class="Box-sc-1bsf784-0 Column-t2p57o-0 dgpVoP"
        width="1"
      >
        <div
          class="Box-sc-1bsf784-0 sc-1k44dfv-0 dCzAzE"
        >
          <div
            class="Box-sc-1bsf784-0 sc-1k44dfv-0 eJJIkK"
          >
            <div
              class="sc-1k44dfv-0 lnlLeU"
              mb="xs,m"
              px="0,default"
              style="text-align: center;"
            >
              

              <h3
                class="sc-1k44dfv-0 eUGlGC"
              >
                My title
              </h3>
            </div>
            <div
              class="Box-sc-1bsf784-0 fmKwic"
            >
              <div
                class="sc-1k44dfv-0 dywxpH"
                mb="0,0"
                style="text-align: center;"
              >
                

                <p>
                  My body text
                </p>
              </div>
            </div>
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 fAJPuz"
            />
          </div>
          <div
            class="fresnel-container fresnel-greaterThanOrEqual-smallDesktop "
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 jYMsma"
            >
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 dYxbmH"
              />
            </div>
          </div>
          <div
            class="fresnel-container fresnel-lessThan-smallDesktop "
          >
            <div
              class="Box-sc-1bsf784-0 sc-1k44dfv-0 jYMsma"
            >
              <div
                class="Box-sc-1bsf784-0 sc-1k44dfv-0 dYxbmH"
              />
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
